### PR TITLE
Changes evidence bags from /obj/item/weapon to /obj/item/weapon/storage

### PR DESCRIPTION
**Pull request by MadmanMartian**
:cl:
 * tweak: Evidence bags are now a child of storage.
 * bugfix: Fixes evidence bags being able to unglue superglued items
 * bugfix: Fixes evidence bags being able to storage supermatter splints safely

### CODE REVIEW

**Comment by DamianX** [comment]
OK but I thought the point of the evidence bag was so you could collect things without getting your fingerprints on them. How are you going to do that without the preattack?

**Comment by ShiftyRail** [comment]
Very opaque change
I think it could remove a lot of "features" associated with evidence bags such as using them to unglue something or to storage supermatter splints safely

If this is the intent of the PR, state so in the body, and add [feature loss]
If it's not, then fine a way to keep those behaviours whilst keeping the file clean

**Comment by jknpj** [comment]
>feature loss
>feature

Don't let your bugs be your features

**Comment by ShiftyRail** [comment]
They're features
The bag is in bluespace plastic

**Comment by MadmanMartian** [comment]
@DamianX That's what the use_to_pickup is for. Emulates the same functionality of 'hit the object with an evidence bag to store it', but with more sanity.

@ShiftyRail Am I really reading that you want to keep exploits in the game because they're of a benefit to you?

**Comment by DamianX** [comment]
👌 then

**Comment by DamianX** [comment]
ackshually, test it, it looks like `handle_item_insertion` (called by `storage/preattack`) calls `add_fingerprint`

**Comment by MadmanMartian** [comment]
Yeah, to the storage bag, not to the item being picked up.

File: code/modules/detectivework/detective_work.dm [hunk 1]
```diff
@@ -171,12 +171,11 @@ var/const/FINGERPRINT_COMPLETE = 6	//This is the output of the stringpercent(pri
...
-				else 
-					if(istype(I, /obj/item/weapon/evidencebag))
-						scanning = I.contents[1]
-						scanning.forceMove(src)
-						I.overlays.len = 0
-						I.icon_state = "evidenceobj"		
+				else
+					if(istype(I, /obj/item/weapon/storage/evidencebag))
+						var/obj/item/weapon/storage/evidencebag/EVB = I
+						scanning = EVB.contents[1]
```
Comment (DamianX) [inline code/modules/detectivework/detective_work.dm:177]: I guess it always used to be like this but won't this runtime if it's empty?

Comment (MadmanMartian) [reply code/modules/detectivework/detective_work.dm:177]: Yeah, it runtimes on live.

**Pull request merged**